### ISSUE TITLE
Syntax-Support for Term Occurrences (MUST, SHOULD, MUST NOT)

### ISSUE THREAD

**Issue by ldegen** [ldegen/irma#5]
The default-operator used when combining multiple words in a query should be configurable. Right now it is always AND.
(This has been addressed, see comments)

Also, it would be really nice to have some syntax to express things like MUST NOT, MUST, SHOULD. 
Right now, only boolean operators are supported, but they are less powerful.
(This is still to be done)

**Comment by ldegen** [comment]
Actually, the default operator being used is configurable as of 0.16.0.
This can always be done per search request by adding `qop=or` to your query string.
With the current version (0.20.0), you can also specify this in your configuration file, although the option is somewhat deeply burried: You need to customize the settings of the ast transformer which is used by the multi-match query component, which in turn is part of the default search semantics.
(We *really* need proper documentation for this!)

So basically, you need something like this:

``` yaml
types:
  my-type:
    seachSemantics: !search-semantics
      # the default search semantics supports specifying a set of query "components".
      # Think of each one as a filter transforming a search request into an elasticsearch query.
      # Per default, two components are used: One for parsing the query string (i.e. the value of the 
      # `q`-URL-Parameter) and one for processing filters. We are interessted in the first one,
      # which parses the query string into an AST and than transforms the AST into an Elasticsearch-
      # specific "Multi-Match"-Query.
      queryComponents:
        - !multi-match-query
          transformer: !ast-transformer
            defaultOpterator: or

        # do not forget the second query component, otherwise your filters will not work
        - !filter-query
```

**Comment by ldegen** [comment]
The syntax currently does not support occurrences (MUST NOT, MUST, SHOULD). I *think* it should be possible to extend the query syntax such that it models occurrences as unary operators, similar to "NOT". In fact, the current NOT very much works (both syntactically and semantically) as  the "MUST NOT" would work. We could do the same for MUST and SHOULD.

I will rename this ticket, since the remaining TODOs are specific to the query string parser.

**Comment by ldegen** [comment]
update: Support for occurences (should, +must, -must_not) landed in 0.32.1
